File: Stack/1935-후위표기식2.py
N = int(input())
expression = input()
stack = []
operand = {}

for i in expression:
    if 'A' <= i <= 'Z' and not i in operand:
        operand[i] = int(input())

for i in expression:
    if 'A' <= i <= 'Z':
        stack.append(operand[i])

    elif i == '*':
        b = stack.pop()
        a = stack.pop()
        stack.append(a*b)

    elif i == '+':
        b = stack.pop()
        a = stack.pop()
        stack.append(a + b)

    elif i == '/':
        b = stack.pop()
        a = stack.pop()
        stack.append(a/b)

    elif i == '-':
        b = stack.pop()
        a = stack.pop()
        stack.append(a-b)
    
print(f'{stack[-1]:.2f}')

File: Stack/1935-후위표기식2.cpp
#include <iostream>
#include <stack>
using namespace std;

int main()
{
    int N;
    char ch;

    string expression;

    stack<int> s;
    cin>>N;
    cin>>expression;

    for(int i=0; i<N; i++){
        ch = (char)getchar();

        cout<<ch<<endl;
    }

}